direct/utils/__init__.py

```diff
@@ -535,3 +535,59 @@ def dict_flatten(in_dict: DictOrDictConfig, dict_out: Optional[DictOrDictConfig]
             continue
         dict_out[k] = v
     return dict_out
+
+
+def reshape_array_to_shape(array: np.ndarray, requested_shape: Tuple[int, ...]) -> np.ndarray:
+    """Reshapes the given array to match the requested shape by adding dimensions of size 1 where necessary.
+
+    Parameters
+    ----------
+    array : np.ndarray
+        The input array to be reshaped.
+    requested_shape tuple of ints
+        The desired shape of the output array.
+
+    Returns
+    -------
+    np.ndarray
+        The reshaped array with the requested shape.
+
+    Example
+    -------
+    >>> array1 = np.random.rand(4, 5)
+    >>> requested_shape1 = (4, 5, 1)
+    >>> result1 = reshape_array_to_shape(array1, requested_shape1)
+    >>> print(result1.shape)  # Output: (4, 5, 1)
+
+    >>> array2 = np.random.rand(4, 5)
+    >>> requested_shape2 = (1, 4, 5, 1)
+    >>> result2 = reshape_array_to_shape(array2, requested_shape2)
+    >>> print(result2.shape)  # Output: (1, 4, 5, 1)
+
+    >>> array3 = np.random.rand(2, 4, 5)
+    >>> requested_shape3 = (2, 4, 5, 1)
+    >>> result3 = reshape_array_to_shape(array3, requested_shape3)
+    >>> print(result3.shape)  # Output: (2, 4, 5, 1)
+    """
+
+    # Get the current shape of the array
+    current_shape = array.shape
+
+    # Check if the current shape already matches the requested shape
+    if current_shape == requested_shape:
+        return array
+
+    # Initialize a new shape list with ones
+    new_shape = [1] * len(requested_shape)
+
+    # Fill in the new shape list with dimensions from the current shape where appropriate
+    j = 0  # Index for current shape
+    for i, dim in enumerate(requested_shape):
+        if j < len(current_shape) and dim == current_shape[j]:
+            new_shape[i] = current_shape[j]
+            j += 1
+
+    # Reshape the array to the new shape
+    reshaped_array = np.reshape(array, new_shape)
+
+    return reshaped_array
```

Comment:
I am not sure about this function as the behavior is quite difficult to understand if you read it inline. Is there a reason why you can't do the reshaping in the locations where required? It will make it easier. Often you can just do something like [:, :, np.newaxis, :] or so.

Reply:
So basically this is used for reshaping the masks. In most cases you might have a produced mask of shape (x, y) and want to reshape it (1 for coils, x, y, 1 for channels) for requested shape (coils, x, y, channels). But you may have other cases such as  requested shape (coils, z, x, y, channels) where the desired reshape is (1, 1, x, y, 1) for a generated mask (x, y), and so on.

The reason I opted out of [:, :, np.newaxis, :] is so that we don't need to check in every mask function (we have like > 10) whether the input shape is 2D or 3D and whether the requested mask is static or dynamic. Any combination of these would expect a different addition of an axis.